### DISCUSSION TITLE
Rendering custom formats within projects

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#7452] [Q&A] [answered]
### Description

I'm running into unexpected behavior when rendering a file that uses a custom format that is installed at the project level. Say you have a default website with one new file, `slides/deck1.qmd` and a custom revealjs format, `storybook`, installed at the project level.

```
├── _quarto.yml
├── about.qmd
├── index.qmd
├── styles.css
├── _extensions
|   ├── mcanouil
|       ├── storybook
|           ├── _extension.yml
|           ├── ... and so on...
├── slides
|   ├── desk1.qmd
```

When I open up `desk1.qmd` in RStudio and click the Render button, I get:

``` bash
ERROR: NotFound: No such file or directory (os error 2), stat '/Users/andrewbray/Desktop/testwebsite/slides/_extensions/mcanouil/storybook/mcsb.svg'

NotFound: No such file or directory (os error 2), stat '/Users/andrewbray/Desktop/testwebsite/slides/_extensions/mcanouil/storybook/mcsb.svg'
```

Which makes it look like it's looking for the extension at the document level. It works just fine if I render the entire project or if I run `quarto render slides/deck1.qmd`. Is this the intended behavior?

MWE found here:

<https://github.com/andrewpbray/testwebsite>

### Versions
I'm running quarto 1.3.433 and RStudio 2023.09.1+494 (2023.09.1+494).

**Comment by mcanouil** [comment]
Is "slides" in the extension folder?

**Reply by andrewpbray** [reply]
Ah, i misaligned that dir. It's now revised in the same structure as can be found in the MWE.

Aside: what's the cleanest way to format a dir in gfm or whatever if being used in these threads? I've just been copying and pasting that funny bar-hyphen character from elsewhere into the thread - not even sure what it is called.

**Reply by mcanouil** [reply]
By the way, why not sharing a Git repository?

**Reply by andrewpbray** [reply]
Happy to do that if that's preferred. Months ago I noticed Dragonstyle doing back and forth with zips and wondered if that was preferred.

**Reply by mcanouil** [reply]
He might prefer, but I'd rather have a Git repository which can produce a zip archive anyway but allow to quickly see the files.

See the "Download ZIP" button
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/6fddbaa9-f9c5-413a-b576-31587c8a3bb8">

**Reply by andrewpbray** [reply]
One moment...

**Reply by andrewpbray** [reply]
Link now in original post.

**Reply by mcanouil** [reply, marked as answer]
It seems to be an issue with RStudio.
Note that I am running the latest development version of Quarto.

https://github.com/quarto-dev/quarto-cli/assets/8896044/e28a609a-6830-4fa4-83e2-ca90db06415d

**Reply by andrewpbray** [reply]
Thanks for testing that. Should I open an issue on the RStudio IDE repo?

**Reply by mcanouil** [reply]
First, you should check with latest pre-release and possibly also on VSCode in your setup just in case.
Then, yes open an issue on RStudio IDE repository as the execute context seems off.

**Reply by andrewpbray** [reply]
The issue was posted here:

[https://github.com/rstudio/rstudio/issues/13961](https://github.com/rstudio/rstudio/issues/13961)

**Reply by andrewpbray** [reply]
@mcanouil Unfortunately there was no take-up on the RStudio IDE issue - not assigned any labels or anything. Does this indicate this won't be fixed? (and therefore I should just find a workaround?).

**Reply by mcanouil** [reply]
I am not working on anything related to the RStudio IDE.
They have their way of working.
There is no point to ask this here.

**Reply by dragonstyle** [reply]
This just came all the way back to Quarto - the issue appears to me to be a Quarto issue. I can reproduce the problem by cloning the repo and previewing deck1.qmd from within the slide directory in a shell. It's possible that the Quarto extension in vs.code is doing something clever with the working dir, but we definitely don't appear to be resolving this logo correctly for some reason.

I'm going to convert this to a Quarto issue.

**Reply by dragonstyle** [reply]
(opened here: https://github.com/quarto-dev/quarto-cli/issues/8385)
